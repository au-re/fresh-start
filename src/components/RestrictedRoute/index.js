import { Redirect, Route } from "react-router-dom";

import React from "react";

/**
 * Restricted Route, handles logic for redirection
 *
 * @param {any} { component: Component, isLoggedIn, redirectTo, ...rest  }
 * @return {object} Restricted Route
 */
const RestrictedRoute = ({ component: Component, isLoggedIn, redirectTo, ...rest  }) =>
  <Route
    {...rest}
    render={(props) =>
      isLoggedIn
        ? <Component {...props} />
        : <Redirect
          to={{
            pathname: redirectTo,
            state: { from: props.location }
          }}
        />} />;

export default RestrictedRoute;
