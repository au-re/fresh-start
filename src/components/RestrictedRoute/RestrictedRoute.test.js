import Enzyme, { mount, shallow } from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import React from "react";
import RestrictedRoute from "./index.js";
import { Router } from "react-router";
import createBrowserHistory from "history/createBrowserHistory";

Enzyme.configure({ adapter: new Adapter() });

describe("RestrictedRoute", () => {

  const history = createBrowserHistory();

  it("fails without prop component", () => {
    expect(mount(
      <Router history={history}>
        <RestrictedRoute isLoggedIn component={() => <div/>} />
      </Router>)).toMatchSnapshot();
  });
});
