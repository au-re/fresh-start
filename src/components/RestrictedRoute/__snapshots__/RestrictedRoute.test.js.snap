// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestrictedRoute fails without prop component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <RestrictedRoute
    component={[Function]}
    isLoggedIn={true}
  >
    <Route
      render={[Function]}
    >
      <component
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
        location={
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          }
        }
        match={
          Object {
            "isExact": true,
            "params": Object {},
            "path": "/",
            "url": "/",
          }
        }
      >
        <div />
      </component>
    </Route>
  </RestrictedRoute>
</Router>
`;
